### PR TITLE
[FIRRTL] Tweak naming

### PR DESCRIPTION
**Pull request by darthscsi**
Lots of _T_ and _WIRE were appearing in the output.  These are quite useless.  We filter these out at all name preservation levels.  This also allows putting other names which start with '_' as droppable.

Also reorder a pass to increase parallelism [NFCI]

### CODE REVIEW

**Comment by darthscsi** [comment]
@mikeurbach , @uenoku I think this might filter out a lot of the truely useless names which are making it to the output, without harming much else.  I was seeing lots of _T_ coming out of firtool.  This seems to clean it up.  Hopefully should improve CSE for resets on verif constructs.

File: lib/Dialect/FIRRTL/Transforms/DropName.cpp [hunk 1]
```diff
@@ -24,26 +24,59 @@ struct DropNamesPass : public DropNameBase<DropNamesPass> {
...
+      });
+    } else if (preserveMode == PreserveValues::All) {
+      // Drop the name if it isn't considered meaningful.
+      dropNamesIf(namesChanged, namesDropped, [](FNamableOp op) {
+        if (isUselessName(op.getName()))
+          return ModAction::Demote;
+        return ModAction::Keep;
+      });
+      numNamesConverted += namesChanged;
+      numNamesDropped += namesDropped;
```
Comment (uenoku) [inline lib/Dialect/FIRRTL/Transforms/DropName.cpp:58]: Wrong place?

File: lib/Dialect/FIRRTL/FIRRTLFolds.cpp [hunk 2]
```diff
@@ -1797,9 +1815,13 @@ struct FoldNodeName : public mlir::RewritePattern {
...
-        rewriter.updateRootInPlace(newOp,
-                                   [&] { newOp->setAttr("name", name); });
+      auto newName = name.getValue();
+      if (newOpName)
+        newName = chooseName(newOpName.getValue(), name.getValue());
+      if (!newOpName || newOpName.getValue() != newName)
+        rewriter.updateRootInPlace(newOp, [&] {
+          newOp->setAttr("name", rewriter.getStringAttr(newName));
+        });
     }
```
Comment (uenoku) [inline lib/Dialect/FIRRTL/FIRRTLFolds.cpp:1825]: Could you do the same for `replaceOpAndCopyName`?

**Review by uenoku — APPROVED** [review]
LGTM

**Pull request merged**